pkg/virt-controller/services/renderresources.go

```diff
@@ -690,3 +702,31 @@ func multiplyMemory(mem resource.Quantity, multiplication float64) resource.Quan
 	mem.Add(*additionalOverhead)
 	return mem
 }
+
+func getMemoryLimitsRatio(namespace string, namespaceStore cache.Store) float64 {
+	if namespaceStore == nil {
+		return DefaultMemoryLimitOverheadRatio
+	}
+
+	obj, exists, err := namespaceStore.GetByKey(namespace)
+	if err != nil {
+		log.Log.Errorf("Error retrieving namespace from informer. Using the default memory limits ratio. %s", err.Error())
+		return DefaultMemoryLimitOverheadRatio
+	} else if !exists {
+		log.Log.Errorf("namespace %s does not exist. Using the default memory limits ratio.", namespace)
+		return DefaultMemoryLimitOverheadRatio
+	}
+
+	ns := obj.(*k8sv1.Namespace)
```

Comment:
Might be worth checking if the cast worked?
```suggestion
	ns, ok := obj.(*k8sv1.Namespace)
	if !ok {
		<log err and return default>
	}
```

Reply:
You might be surprised to see the number of times we do vs. the number of times we don't do it. :slightly_smiling_face:

Reply:
:) well, being an object that comes directly from a successful `get` on the namespace store, it is quite safe to assume it'll cast...
But better safe than sorry, the highly improbable failure case will now result in a warning instead of a segfault!

Reply:
Sure thing :)

Reply:
Converted to `Error`. Thank you!